main/engines/users.py

```diff
@@ -0,0 +1,34 @@
+import bcrypt
+from sqlalchemy import select
+
+from main import db
+from main.models.user import UserModel
+
+
+async def add_user(email: str, password: str) -> UserModel:
+    user = UserModel(
+        email=email,
+        hashed_password=bcrypt.hashpw(
+            password.encode(),
+            bcrypt.gensalt(),
+        ),
+    )
+
+    db.session.add(user)
+    await db.session.commit()
+
+    return user
+
+
+async def get_user_by_email(email: str) -> UserModel:
+    statement = select(UserModel).filter(UserModel.email == email)
```

Comment:
Let's use `.where` instead of `.filter`: https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-orm-usage

Reply:
![Uploading image.png…]()

Reply:
No, `.where/.filter` is better than `.filter_by` in virtually every aspect